### PR TITLE
Build corrections

### PR DESCRIPTION
**Pull request by BorisMansencal**
*Thanks a lot for contributing to DGtal, before submitting your PR, please fill up the description and make sure that all checkboxes are checked. Please remove these lines in your PR.*

# PR Description

This PR do some small build corrections.
In particular, code was not building with gcc 9.1.1 on Fedora 30 : due to missing include of <map> in src/DGtal/helpers/Parameters.h.
It also correct warning -Wshadow on src/DGtal/helpers/Shortcuts.h

However, there are still some warnings when building in Debug mode:
- On Fedora 30, with gcc 9.1.1 : lots of warnings due to -Wdeprecated-copy 
- On Ubuntu 19.04, with clang 8.0.0, 4 warnings due to -Wliteral-conversion on tests/shapes/testAstroid2D.cpp:
```
/home/mansenca/DGtal/tests/shapes/testAstroid2D.cpp:112:50: warning: implicit conversion from 'double' to 'int' changes
      value from 2.220446049250313E-16 to 0 [-Wliteral-conversion]
      REQUIRE_THAT( res, Catch::WithinULP(M_PI_2,DBL_EPSILON) || Catch::WithinULP(3*M_PI_2,DBL_EPSILON) );
                         ~~~~~                   ^~~~~~~~~~~
/usr/include/clang/8.0.0/include/float.h:133:21: note: expanded from macro 'DBL_EPSILON'
#define DBL_EPSILON __DBL_EPSILON__
                    ^~~~~~~~~~~~~~~
<built-in>:161:25: note: expanded from here
#define __DBL_EPSILON__ 2.2204460492503131e-16
                        ^~~~~~~~~~~~~~~~~~~~~~
/home/mansenca/DGtal/tests/catch.hpp:16592:75: note: expanded from macro 'REQUIRE_THAT'
#define REQUIRE_THAT( arg, matcher ) INTERNAL_CHECK_THAT( "REQUIRE_THAT", matcher, Catch::ResultDisposition::Normal, arg )
                                                                          ^~~~~~~
/home/mansenca/DGtal/tests/catch.hpp:3671:74: note: expanded from macro 'INTERNAL_CHECK_THAT'
            catchAssertionHandler.handleExpr( Catch::makeMatchExpr( arg, matcher, #matcher##_catch_sr ) ); \
                                                                         ^~~~~~~
``` 

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/helpers/Shortcuts.h [hunk 1]
```diff
@@ -1896,12 +1896,14 @@ namespace DGtal
...
 	  // Simplify materials (very useful for blender).
-          Idx j = 0;
           std::map<Color,Idx> map_colors;
-          for ( auto && c : diffuse_colors )
-            if ( ! map_colors.count( c ) )
-              map_colors[ c ] = j++;
-
+	  {
+            Idx j = 0;
+            for ( auto && c : diffuse_colors )
```
Comment (copyme) [inline src/DGtal/helpers/Shortcuts.h]: @JacquesOlivierLachaud is not that in DGtal the convention is to name instances and variables in camel case? Anyways, this is a general question for these files and not for the PR itself.

Comment (JacquesOlivierLachaud) [reply src/DGtal/helpers/Shortcuts.h]: Camel case is for classes and methods. I don't remember that we had camel case conventions for local variables and I would not be in favour of it.

**Comment by JacquesOlivierLachaud** [comment]
Do you want to update the ChangeLog Boris ?

**Comment by JacquesOlivierLachaud** [comment]
Or are you waiting to push other changes ?

**Comment by BorisMansencal** [comment]
I would like to correct remaining warnings with gcc 9.1.1.
So I should push new changes and then update the ChangeLog.

**Comment by BorisMansencal** [comment]
I think clang warns about a real bug on tests/shapes/testAstroid2D.cpp:
`Catch::WithinULP(M_PI_2,DBL_EPSILON)` is invalid: WithinULP needs an `int` as second argument. 
A fix is to replace `WithinULP` by `WithinAbs`

**Comment by BorisMansencal** [comment]
Ok, I'm done.
With this PR, DGtal should compile in Debug mode and WIT_ITK=ON without any warning with gcc 9.1.1 and clang 8.0 & 9.0.

File: tests/topology/testHalfEdgeDataStructure.cpp [hunk 2]
```diff
@@ -47,6 +47,9 @@ typedef HalfEdgeDataStructure::PolygonalFace    PolygonalFace;
 typedef HalfEdgeDataStructure::Edge             Edge;
 typedef HalfEdgeDataStructure::Arc              ArcT; //Arc already defined in wingdi.h
 typedef HalfEdgeDataStructure::VertexIndexRange VertexIndexRange;
+typedef HalfEdgeDataStructure::Size Size;
```
Comment (hadjiszs) [inline tests/topology/testHalfEdgeDataStructure.cpp]: I would suggest `using Foo::Bar` rather than `typedef Foo::Bar Bar` to let the ADL doing the job

Comment (rolanddenis) [reply tests/topology/testHalfEdgeDataStructure.cpp]: Is ADL not only related to function name lookup ?

**Pull request merged**